cdap-common/src/test/java/io/cdap/cdap/common/metadata/QueryParserTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.metadata;
+
+import io.cdap.cdap.common.metadata.QueryTerm.Qualifier;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.List;
+
+public class QueryParserTest {
+  @Test
+  public void testBasicParse() {
+    String inputQuery = "tag1 tag2 +tag3 tag4+";
+    List<QueryTerm> targetOutputQueryTerms = new ArrayList<>();
+    targetOutputQueryTerms.add(new QueryTerm("tag1", Qualifier.OPTIONAL));
+    targetOutputQueryTerms.add(new QueryTerm("tag2", Qualifier.OPTIONAL));
+    targetOutputQueryTerms.add(new QueryTerm("tag3", Qualifier.REQUIRED));
+    targetOutputQueryTerms.add(new QueryTerm("tag4+", Qualifier.OPTIONAL));
+    List<QueryTerm> actualOutputQueryTerms = QueryParser.parse(inputQuery);
+
+    for (int i = 0; i < 4; i++) {
```

Comment:
This would miss the case where the actual terms have a 5th element. I think it is more concise to use 
```
Assert.assertEquals(targetOutputQueryTerms, actualOutputQueryTerms);
```

Reply:
Fixed.